imslp.js

```diff
@@ -0,0 +1,239 @@
+	var rows = doc.querySelectorAll('.categorypagelink[href]');
+	for (var i = 0; i < rows.length; i++) {
+		var href = rows[i].href;
+		var title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+async function doWeb(doc, url) {
+	if (detectWeb(doc, url) == 'multiple') {
+		let items = await Zotero.selectItems(getSearchResults(doc, false));
+		if (!items) return;
+		for (var urls of Object.keys(items)) {
+			await scrape(await requestDocument(urls));
+		}
+		//Z.debug(ZU.processDocuments(Object.keys(items)), scrape)
+	}
+	else {
+		scrape(doc, url);
+	}
+}
+
+async function scrape(doc, url = doc.location.href) {
+	var item = new Zotero.Item("book");
+	var title = text(doc, '#firstHeading');
+	
+	//check metadata
+	var rows = doc.querySelector('.wi_body table').rows;
+	for (var i = 2; i < rows.length; i++) {
+		//Z.debug(rows[i].cells[0].innerText);
+		//Z.debug(rows[i].cells[1].innerText);
+		if (rows[i].cells[0].innerText.trim().includes("Name Translations")) {
+			var names = rows[i].cells[1].innerText.trim();
+		}
+		if (rows[i].cells[0].innerText.trim() == "Composer") {
+			var author = rows[i].cells[1].innerText.trim();
+		}
+		if (rows[i].cells[0].innerText.trim().includes("Opus/Catalogue Number")) {
+			var seriesNumber = rows[i].cells[1].innerText.trim();
+		}
+		if (rows[i].cells[0].innerText.trim() == "Key") {
+			var key = rows[i].cells[1].innerText.trim();
+		}
+		if (rows[i].cells[0].innerText.trim() == "Text Incipit") {
+			var incipit = rows[i].cells[1].innerHTML;
+		}
+		if (rows[i].cells[0].innerText.trim().includes("Movements/Sections")) {
+			var movements = rows[i].cells[1].innerHTML;
+		}
+		if (rows[i].cells[0].innerText.trim().includes("Year/Date of Composition")) {
+			var date = rows[i].cells[1].innerText.trim();
+		}
+		if (rows[i].cells[0].innerText.trim().includes("First Performance")) {
+			var place = rows[i].cells[1].innerText.trim();
+		}
+		if (rows[i].cells[0].innerText.trim().includes("First Publication")) {
+			var publisher = rows[i].cells[1].innerText.trim();
+		}
+		if (rows[i].cells[0].innerText.trim() == "Language") {
+			var language = rows[i].cells[1].innerText.trim();
+		}
+		if (rows[i].cells[0].innerText.trim().includes("Average Duration")) {
+			var duration = rows[i].cells[1].innerText.trim();
+		}
+		if (rows[i].cells[0].innerText.trim() == "Piece Style") {
+			var style = rows[i].cells[1].innerText.trim();
+		}
+
+		/* The metadata of instrumention is too mess.
+		if (rows[i].cells[0].innerText.trim() == "Instrumentation") {
+			var instrumentation = rows[i].cells[1].innerText.trim();
+		}*/
+		//Z.debug(instrumentation);
+	}
+
+	//push metadata
+	item.title = title.replace(/\(.*\)/, "");
+	item.url = url;
+	item.libraryCatalog = null;
+	if (author) {
+		item.creators.push(ZU.cleanAuthor(author, "author", true));
+		item.tags.push(author.split(',')[0]);
```

Comment:
I don't think adding the author as a tag makes sense -- they're already in the metadata as author.